### PR TITLE
Query bookings which are not in the past

### LINKED ISSUE #103: Query bookings which are after or on the current date only
at one point, bookings will take too much space on homepage, and I think it would be irrelevant to show bookings in the past as well.

### PR DESCRIPTION
**Pull request by welvin21**
Closes #103

### CODE REVIEW

**Comment by utkarsh867** [comment]
@welvin21 I think we have to take another route to solve this problem.

I propose changing phoenix queries to return the bookings only after a certain date and time instead of doing this in the front-end manually.

We can think of this from the point of view of Prisma, that prisma is still going to have to query all the data anyway, and then graphql does the filter. I guess it's not that amazing.

File: src/gql/bookings.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { gql } from 'apollo-boost';
 
 export const ROOM_BOOKINGS = gql`
-  query bookings($room: String!) {
-    bookings(data: { room: { number: $room } }) {
+  query bookings($room: String!, $start: String) {
+    bookings(data: { room: { number: $room } }, start: $start) {
```
Comment (utkarsh867) [inline src/gql/bookings.js]: Correct me if I am wrong.

Specifying a `start` for the booking query will try to match the bookings that have the specific `start` time. I do not think this is what we want? We want all the bookings after `start` right?

Comment (welvin21) [reply src/gql/bookings.js]: @utkarsh867 no actually, `start` will be passed to phoenix, and phoenix will use `start` as `start_gte` param for prisma. So in general, it returns all the bookings after `start`

**Comment by utkarsh867** [comment]
Refer to rcltech/phoenix#67 for changes related to this commit.

I think we need a fix there as well.

**Comment by welvin21** [comment]
> @welvin21 I think we have to take another route to solve this problem.
> 
> I propose changing phoenix queries to return the bookings only after a certain date and time instead of doing this in the front-end manually.
> 
> We can think of this from the point of view of Prisma, that prisma is still going to have to query all the data anyway, and then graphql does the filter. I guess it's not that amazing.

@utkarsh867 ohh i see, so according to this implementation, there won't be any dynamic parameter passed from `owl` that will decide when the `start` time filter of the bookings is. And that filter will completely be fixed on `phoenix`? (let's say `phoenix` will always return all the bookings after or equal to today's date)

**Comment by utkarsh867** [comment]
> > @welvin21 I think we have to take another route to solve this problem.
> > I propose changing phoenix queries to return the bookings only after a certain date and time instead of doing this in the front-end manually.
> > We can think of this from the point of view of Prisma, that prisma is still going to have to query all the data anyway, and then graphql does the filter. I guess it's not that amazing.
> 
> @utkarsh867 ohh i see, so according to this implementation, there won't be any dynamic parameter passed from `owl` that will decide when the `start` time filter of the bookings is. And that filter will completely be fixed on `phoenix`? (let's say `phoenix` will always return all the bookings after or equal to today's date)

I think we can still keep that functionality? For example, when a user is making a booking for the future, the user wants to know bookings only for that booking day or later to schedule the event without overlap. But in general, Phoenix should be modified to not query events that have happened in the past unless it is specifically asked for (for example, when showing user profile, we need all past, present and future events of the user).

**Comment by welvin21** [comment]
@utkarsh867 I will make some changes for this and rcltech/phoenix#67 shortly. Just confirming regarding the implementation, `phoenix` by default should return present and future bookings. And it will return past bookings if specified only right (by passing something like `start` variable in the query) ??

**Comment by utkarsh867** [comment]
@welvin21 Aren't we missing something here?

Consider a booking where it is still on-going. Hence, it is not available for booking. However, it will also not be returned by phoenix because the start time may be in the past.

**Comment by welvin21** [comment]
> @welvin21 Aren't we missing something here?
> 
> Consider a booking where it is still on-going. Hence, it is not available for booking. However, it will also not be returned by phoenix because the start time may be in the past.

@utkarsh867 ohh rightt, well i should change `phoenix` then. i think we can add `end_gte: new Date()` to prisma query param for default `bookings` and `events` queries. do you see any issue with that implementation?

**Comment by 98sean98** [comment]
@welvin21 is this PR still needed? I think we made the fix to phoenix already right? Or are there additional changes that are more relevant?

**Comment by welvin21** [comment]
> @welvin21 is this PR still needed? I think we made the fix to phoenix already right? Or are there additional changes that are more relevant?

@98sean98 yeah `phoenix` will resolve present and future bookings by default already. But `start_limit` should be there as an optional parameter (in case we want to query past bookings as well). And there's a couple of changes in export/import as well in this PR, just some `index.js`s.

**Review by 98sean98 — APPROVED** [review]

**Pull request merged**